### PR TITLE
Further refactoring of upgrade tests and removal of etcd upgrade tests which isn't even run

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/98326

```release-note
NONE
```

/kind cleanup
/priority important-soon

### CODE REVIEW

File: test/e2e/cloud/gcp/cluster_upgrade.go [hunk 1]
```diff
@@ -146,35 +135,13 @@ var _ = ginkgo.Describe("Downgrade [Feature:Downgrade]", func() {
 	})
 })
 
-var _ = ginkgo.Describe("etcd Upgrade [Feature:EtcdUpgrade]", func() {
```
Comment (wojtek-t) [inline test/e2e/cloud/gcp/cluster_upgrade.go:149]: @spiffxp - can you please double check me on this?

I was trying to find where this tests is running and failed to find anything.

Comment (spiffxp) [reply test/e2e/cloud/gcp/cluster_upgrade.go:149]: I also couldn't find anything

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**